src/js/workers/hex_parser.js

```diff
@@ -77,7 +77,8 @@ function read_hex_file(data) {
                 }
                 break;
             case 0x04: // extended linear address record
-                extended_linear_address = (parseInt(content.substr(0, 2), 16) << 24) | parseInt(content.substr(2, 2), 16) << 16;
+                // input address is UNSIGNED
+                extended_linear_address = ((parseInt(content.substr(0, 2), 16) << 24) | parseInt(content.substr(2, 2), 16) << 16) >>> 0;
```

Comment:
Not sure why you are adding the `>>> 0` here?

Reply:
Byte overflow? not sure on the right term. I checked with the binary he has posted:
```
Did console.log of extended_linear_address when it was set.
Hex conversion is what windows calculator says.
Old way: -1748107264    FFFF FFFF 97CE 0000
New way:  2546860032              97CE 0000
```
I got suggested this when I looked for 'uint32 javascript' so it seems reasonable.
https://stackoverflow.com/questions/22335853/hack-to-convert-javascript-number-to-uint32

Reply:
yes, it's because all bitwise operators in JS are 32 bit *signed*.  The >>> 0  forces the result to UNSIGNED.

Reply:
Heh, signed bitwise operators, who wouldn't want those.